core/src/actions/plugins.ts

```diff
@@ -81,12 +85,33 @@ export class PluginUninstall extends AuthenticatedAction {
     this.permission = { topic: "system", mode: "write" };
     this.inputs = {
       plugin: { required: true },
+      restart: { required: false, default: "false" },
     };
     this.outputExample = {};
   }
 
   async runWithinTransaction({ params }) {
     const response = await Plugins.uninstall(params.plugin);
-    return response;
+
+    if (!response.success) return { success: false };
+    // Return if did not ask to restart
+    if (!params.restart) return { success: response.success };
+    // Otherwise, return, then restart the server.
+    setTimeout(() => {
+      clearPluginConfigCache();
+      api.process.restart();
+    }, restartSleepTime);
+    return { success: response.success, checkIn: restartSleepTime * 4 };
   }
 }
+
+function clearPluginConfigCache() {
+  const files = ["plugins.js"];
+
+  files.forEach((file) => {
+    const fullPath = path.join(__dirname, "..", "config", file);
+    if (fs.existsSync(fullPath)) {
+      delete require.cache[require.resolve(fullPath)];
```

Comment:
`core/(src/dis)/config/plugins` was loaded once at boot and exported the list of plugins at that time.  Node caches improt/require values and assumes they don't change... but in this case, they do!